pkg/publish/default.go

```diff
@@ -203,7 +207,7 @@ func (d *defalt) Publish(ctx context.Context, br build.Result, s string) (name.R
 	// https://github.com/google/go-containerregistry/issues/212
 	s = strings.ToLower(s)
 
-	ro := []remote.Option{remote.WithAuth(d.auth), remote.WithTransport(d.t), remote.WithContext(ctx), remote.WithUserAgent(d.userAgent)}
+	ro := []remote.Option{remote.WithAuthFromKeychain(d.keychain), remote.WithTransport(d.t), remote.WithContext(ctx), remote.WithUserAgent(d.userAgent)}
```

Comment:
Hmm... So now we have the case where a client may have used `publish.WithAuth`, and with this change we may be dropping it on the floor.

I think what we should do is check whether `d.auth == authn.Anonymous`, and only if it is, add `remote.WithAuthFromKeychain(d.keychain)` -- otherwise, keep `remote.WithAuth(d.auth)`. We might want to error if both are passed too, to avoid confusion.

This should only affect clients like Skaffold that use this code as a library though.

cc @halvards

Reply:
I fixed it, it should be okay right now.

Reply:
Thanks for keeping ko-as-library users in mind :slightly_smiling_face:

Reply:
kindly ping @imjasonh ☝️